src/main/java/com/commercetools/sync/products/utils/ProductVariantUpdateActionUtils.java

```diff
@@ -0,0 +1,116 @@
+     * which expected to be removed first.
+     *
+     * @param oldProduct old product with variants
+     * @param newProduct new product draft with variants <b>with resolved references prices references</b>
+     * @return list of update actions to add new variants.
+     * @see #buildRemoveVariantUpdateAction(Product, ProductDraft)
+     */
+    @Nonnull
+    public static List<UpdateAction<Product>> buildAddVariantUpdateAction(@Nonnull final Product oldProduct,
+                                                                          @Nonnull final ProductDraft newProduct) {
+        final List<ProductVariantDraft> newVariants = newProduct.getVariants();
+        final Set<String> oldVariantKeys =
+                collectionToSet(oldProduct.getMasterData().getStaged().getVariants(), ProductVariant::getKey);
+
+        return filterCollection(newVariants, newVariant -> !oldVariantKeys.contains(newVariant.getKey()))
+                .map(ProductVariantUpdateActionUtils::buildAddVariantUpdateActionFromDraft)
+                .collect(toList());
+    }
+
+    /**
+     * Create update action, if {@code newProduct} has {@code #masterVariant#key} different than {@code oldProduct}
+     * staged {@code #masterVariant#key}.
+     *
+     * <p>If update action is created - it is created of
+     * {@link ProductVariantDraft newProduct.getMasterVariant().getSku()}
+     *
+     * @param oldProduct old product with variants
+     * @param newProduct new product draft with variants <b>with resolved references prices references</b>
+     * @return {@link ChangeMasterVariant} if the keys are different.
+     */
+    @Nonnull
+    public static Optional<UpdateAction<Product>> buildChangeMasterVariantUpdateAction(
+            @Nonnull final Product oldProduct,
+            @Nonnull final ProductDraft newProduct) {
+        final String newKey = newProduct.getMasterVariant().getKey();
+        final String oldKey = oldProduct.getMasterData().getStaged().getMasterVariant().getKey();
+        return buildUpdateAction(newKey, oldKey,
+            // it might be that the new master variant is from new added variants, so CTP variantId is not set yet,
+            // thus we can't use ChangeMasterVariant.ofVariantId()
+            () -> ChangeMasterVariant.ofSku(newProduct.getMasterVariant().getSku()));
+    }
+
+    /**
+     * Factory method to create {@link AddVariant} action from {@link ProductVariantDraft} instance.
+     *
+     * <p>The {@link AddVariant} will include:<ul>
+     *     <li>sku</li>
+     *     <li>keys</li>
+     *     <li>attributes</li>
+     *     <li>prices</li>
+     *     <li>images</li>
+     * </ul>
+     *
+     * @param draft {@link ProductVariantDraft} which to add.
+     * @return new {@link AddVariant} update action with properties from {@code draft}
+     */
+    @Nonnull
+    public static AddVariant buildAddVariantUpdateActionFromDraft(@Nonnull final ProductVariantDraft draft) {
```

Comment:
would be good to add the JVM SDK :)

Reply:
also why `public`?

Reply:
c05f147
made internal to use in the tests.